packages/cli/src/lib/bundler/config.ts

```diff
@@ -128,18 +129,81 @@ export async function createConfig(
     }),
   );
 
-  plugins.push(
-    new HtmlWebpackPlugin({
-      meta: {
-        'backstage-app-mode': options?.appMode ?? 'public',
-      },
-      template: paths.targetHtml,
-      templateParameters: {
-        publicPath,
-        config: frontendConfig,
-      },
-    }),
-  );
+  if (options.moduleFederation?.mode !== 'remote') {
+    plugins.push(
+      new HtmlWebpackPlugin({
+        meta: {
+          'backstage-app-mode': options?.appMode ?? 'public',
+        },
+        template: paths.targetHtml,
+        templateParameters: {
+          publicPath,
+          config: frontendConfig,
+        },
+      }),
+    );
+  }
+
+  if (options.moduleFederation) {
+    plugins.push(
```

Comment:
Are there any backstage packages that should be added here? Maybe not immediately, but thinking things like version bridge or whatnot should be shared?

Other packages that don't like duplicates etc? But thats more of a symptom of a different problem I guess.

Reply:
Generally it's should all be fine from a PoV of things working correctly, we already deal with the situation of duplicated dependencies. Packages like `@backstage/frontend-app-api` should never be included in a plugin or dynamic container in the first place.

I did experiment a bit with for example sharing `@backstage/core-components` as an optimization, but didn't really like the result. It's a huge tradeoff to disable tree-shaking for any shared dependency, and in practice I think it will be better to just optimize the individual containers and be fine with a bit of duplication across them. Definitely an area where we need to iterate and experiment though.